src/main_benchmark.cpp

```diff
@@ -0,0 +1,411 @@
+    }
+}
+
+std::vector<size_t> parseShapes(const std::string& cliInputShapes) {
+    auto inputShapes = ovms::tokenize(cliInputShapes, ';');
+    if (inputShapes.size() != 1) {
+        std::cout << __LINE__ << std::endl;
+        throw std::invalid_argument("Invalid shape argument");
+    }
+    std::string firstShape = inputShapes[0];
+    size_t leftBracket = firstShape.find("[");
+    size_t rightBracket = firstShape.find("]");
+    if ((leftBracket == std::string::npos) ||
+        (rightBracket == std::string::npos) ||
+        (leftBracket > rightBracket)) {
+        std::cout << __LINE__ << std::endl;
+        throw std::invalid_argument("Invalid shape argument");
+    }
+    std::string shapeString = firstShape.substr(leftBracket + 1, rightBracket - leftBracket - 1);
+    auto dimsString = ovms::tokenize(shapeString, ',');
+    std::vector<std::size_t> shape;
+    std::transform(dimsString.begin(), dimsString.end(), std::back_inserter(shape),
+                                   [](const std::string& s) -> std::size_t {
+        auto dimOpt = ovms::stoi64(s);
+        if (!dimOpt.has_value()) {
+            std::cout << __LINE__ << " " << s << std::endl;
+            throw std::invalid_argument("Invalid shape argument");
+        }
+                                   return dimOpt.value(); });
+    return shape;
+}
+
+volatile sig_atomic_t shutdown_request = 0;
+
+static void onInterrupt(int status) {
+    shutdown_request = 1;
+}
+
+static void onTerminate(int status) {
+    shutdown_request = 1;
+}
+
+static void onIllegal(int status) {
+    shutdown_request = 2;
+}
+
+static void installSignalHandlers() {
+    static struct sigaction sigIntHandler;
+    sigIntHandler.sa_handler = onInterrupt;
+    sigemptyset(&sigIntHandler.sa_mask);
+    sigIntHandler.sa_flags = 0;
+    sigaction(SIGINT, &sigIntHandler, NULL);
+
+    static struct sigaction sigTermHandler;
+    sigTermHandler.sa_handler = onTerminate;
+    sigemptyset(&sigTermHandler.sa_mask);
+    sigTermHandler.sa_flags = 0;
+    sigaction(SIGTERM, &sigTermHandler, NULL);
+
+    static struct sigaction sigIllHandler;
+    sigIllHandler.sa_handler = onIllegal;
+    sigemptyset(&sigIllHandler.sa_mask);
+    sigIllHandler.sa_flags = 0;
+    sigaction(SIGILL, &sigIllHandler, NULL);
+}
+
+using shape_t = std::vector<size_t>;
+
+OVMS_InferenceRequest* prepareRequest(const std::string& modelName, uint64_t modelVersion, OVMS_DataType datatype, const shape_t& shape, const std::string& inputName, const void* data) {
+    OVMS_InferenceRequest* request{nullptr};
+    OVMS_InferenceRequestNew(&request, modelName.c_str(), modelVersion);
+    OVMS_InferenceRequestAddInput(request, inputName.c_str(), datatype, shape.data(), shape.size());
+    size_t elementsCount = std::accumulate(shape.begin(), shape.end(), 1, std::multiplies<size_t>());
+    OVMS_InferenceRequestInputSetData(request, inputName.c_str(), data, sizeof(float) * elementsCount, OVMS_BUFFERTYPE_CPU, 0);  // TODO sizeof
```

Comment:
What's wrong with sizeof?

Reply:
When we would like to test other precisions as well we need utility function to provide bytesize from OVMS_DataType. This could be C-api utility function as well.